src/beacon-api/app/websockets/routing.py

```
@@ -0,0 +1,299 @@
+                        # Manually invoke the background task since WebSockets don't trigger the FastAPI BackgroundTasks pipeline automatically!
+                        # This is a critical edge case!
+                        for task_def in bg_tasks.tasks:
+                            asyncio.create_task(task_def.func(*task_def.args, **task_def.kwargs))
+
+                        await websocket.send_json({
+                            "type": "command_execution_started",
+                            "job_id": str(job.id),
+                            "job_type": job.job_type,
+                            "status": job.status
+                        })
+                except HTTPException as e:
```

Ruff (0.15.20)

[error] Undefined name `HTTPException`

(F821)